app.ts

```diff
@@ -0,0 +1,388 @@
+					$('.search-container').append(`
+						<div class="results-box">
+							<p class="message">Invalid Input!</p>
+						</div>`);
+				}
+			} else {
+				$('.results-box').remove();
+			}
+			isFunctionRunning = false;
+		})
+		.catch((error) => {
+			throw error;
+		})
+})
+
+// After the range has been returned to the user. The user can click on it and that will show the range of records on the table. 
+function resultsSelect(event: any) {
+	if (isFunctionRunning) {
+		return;
+	}
+	isFunctionRunning = true;
+	let startID: number;
+	let endID: number;
+	let pageEnd: number;
+	let pageStart: number;
+	return apiManager.getRecordCount()
+		.then((count) => {
+			let idRange = $('.results-select').text();
+			let rangeArray = null;
+			rangeArray = idRange.split('-');
+			$('.results-box').remove();
+			startID = parseInt(rangeArray[0]);
+			endID = parseInt(rangeArray[1]);
+			let maxRecords = recordsPerPage();
+			globals.currentPage = Math.ceil(endID / maxRecords);
+			pageEnd = Math.ceil(globals.currentPage / 10) * 10;
+			pageStart = pageEnd - 9;
+			if (endID === count) {
+				startID = ((globals.currentPage - 1) * maxRecords) + 1;
+				endID = count;
+				globals.firstPage = pageStart;
+				globals.lastPage = pageEnd;
+			}
+		})
+		.then(() => {
+			return pageNumbers(pageStart, pageEnd);
+		})
+		.then(() => {
+			return showRecords(startID, endID);
+		})
+		.catch((error) => {
+			throw error;
+		})
+}
+
+// When adjusting the height and on different screen sizes. This function would responsible for calculating how much records should be displayed based on the height of the window itself. 
+function adjustDisplayedRecords() {
+	if (isFunctionRunning) {
+		return;
+	}
+	isFunctionRunning = true;
+	let screenHeight = $(window).height() as number;
+	if (screenHeight < 68) {
+		screenHeight = 68;
+	}
+	let maxRecords: number;
+	let pageStart: number;
+	let pageEnd: number;
+	return apiManager.getRecordCount()
+		.then((count) => {
+			let maxRecords = Math.floor(parseInt(screenHeight as any) / 68);
+			let inputNumber = input();
+			let length = inputNumber.length as number;
+			let inputNumberInt = parseInt(inputNumber);
+			if (inputNumber === '') {
+				let newCurrentPage = Math.ceil(globals.currentFromID / maxRecords);
+				if (newCurrentPage === 1) {
+					globals.currentFromID = 1;
+				}
+				globals.currentToID = globals.currentFromID + (maxRecords - 1);
+				globals.currentPage = newCurrentPage;
+				let originalID = (Math.floor(maxRecords * globals.currentPage) - (maxRecords - 1));
+				globals.difference = globals.currentFromID - originalID;
+			} else {
+				if (length > 0) {
+					let newCurrentPage = Math.ceil(inputNumberInt / maxRecords);
+					if (globals.currentToID > count) {
+						globals.currentToID = count;
+					}
+					globals.currentToID = newCurrentPage * maxRecords;
+					globals.currentPage = newCurrentPage;
+					globals.currentFromID = (globals.currentPage - 1) * maxRecords + 1;
```

Comment:
This line would never be able to set zero, which is a problem, as we should be able to see the first record.